source/docs/software/commandbased/commands-v3/getting-started.rst

```diff
@@ -0,0 +1,280 @@
+    private void tank(double left, double right) { /* set motor outputs */ }
+    private void stop() { tank(0, 0); }
+    private void resetEncoders() { /* ... */ }
+
+    // Public sensor reading
+    public double getDistanceMeters() { return 0.0; }
+
+    // Public command factory
+    public Command driveForTime() {
+      return run(coroutine -> {
+        tank(0.5, 0.5);
+        coroutine.wait(Seconds.of(2));
+        stop();
+      }).named("Drive 2s");
+    }
+  }
+  ```
+
+.. note::
+   ``Mechanism`` provides ``run(Consumer<Coroutine>)`` to build a command for that mechanism and ``runRepeatedly(Runnable)`` for simple periodic loops where the body runs every cycle.
+
+## Writing Imperative Commands
+
+Here’s a classic “drive a distance” example side-by-side. v3 lets you write the steps in-order with a loop and a yield.
+
+.. rubric:: v3 (coroutines)
+
+.. code-block:: java
+
+  Command driveTenFeet = drivetrain
+    .run(coroutine -> {
+      drivetrain.resetEncoders();
+      while (drivetrain.getDistanceMeters() < 3.048) { // ~10 ft
+        drivetrain.tank(0.5, 0.5);
+        coroutine.yield(); // let the scheduler run others
+      }
+      drivetrain.stop();
+    })
+    .named("Drive 10 ft");
+
+.. rubric:: v2 (equivalent)
+
+.. code-block:: java
+
+  Command driveTenFeet =
+    Commands.runOnce(drivetrain::resetEncoders, drivetrain)
+      .andThen(new RunCommand(() -> drivetrain.tank(0.5, 0.5), drivetrain)
+        .until(() -> drivetrain.getDistanceMeters() >= 3.048))
+      .finallyDo(drivetrain::stop)
+      .withName("Drive 10 ft");
+
+Both versions do the same thing. The v3 version reads like the steps you’d write on a whiteboard, and the ``yield()`` keeps the program responsive.
+
+## Useful Coroutine Helpers
+
+.. tab-set-code::
+
+  ```java
+  // Wait for time
+  command = Command.waitFor(Seconds.of(0.25)).named("Delay 250ms");
+
+  // Wait until a condition is true
+  command = Command.waitUntil(() -> arm.atGoal()).named("Wait For Arm");
+
+  // Await another command (schedule it if needed)
+  Command score = Command.noRequirements().executing(coroutine -> {
+    coroutine.await(drivetrain.driveToPose(...));
+    coroutine.await(arm.moveTo(...));
+    coroutine.await(gripper.release());
+  }).named("Score Piece");
+
+  // Run several, continue when any one finishes (others are canceled)
+  Command pickAny = Command.noRequirements().executing(coroutine -> {
+    coroutine.awaitAny(drivetrain.driveToPose(...), vision.alignToTag(...));
+  }).named("Drive Or Align");
+
+  // Keep running until canceled (great for “hold while button pressed”)
+  Command holdIntake = intake.run(coroutine -> {
+    intake.on();
+    coroutine.park(); // never exits on its own
+  }).whenCanceled(intake::off).named("Hold Intake");
+
+  // Drive repeatedly, but stop automatically when a condition trips
+  Command driveUntilBeamBreak = drivetrain
+    .runRepeatedly(() -> drivetrain.tank(0.4, 0.4))
+    .until(() -> sensors.beamBroken())
```

Comment:
```suggestion
    .until(sensors::beamBroken)
```
Perhaps also rename the method, a broken beam is an implementation detail. The public method name should be closer to the problem space.